plugin/samples/authorized-type.yml

```diff
@@ -0,0 +1,42 @@
+---
+kind: pipeline
+type: kubernetes
+name: default
+
+steps:
+  - name: go-test
+    image: golang:1.16
+    commands:
+      - make go-test
+
+  - name: build
+    image: golang:1.16
+    environment:
+      USERNAME:
+        from_secret: username
+      PASSWORD:
+        from_secret: password
+    commands:
+      - make build
+
+trigger:
+  event:
+    exclude:
+      - promote
+  branch:
+    - master
+
+---
```

Comment:
It seems that the `rego` library does not understand a multi-document YAML file, so anything beyond the first document is simply being dropped during json marshaling. You can simply change this document to `type: docker` and see that tests still pass.

Reply:
Very good point, updated the plugin to validate one YAML document at a time.